src/app/core/teams/teams.service.js

```diff
@@ -19,628 +19,620 @@ const
-
-		// Just need to make sure we find one active admin who isn't this user
-		const adminFound = results.some((u) => {
-			const role = getActiveTeamRole(u, team);
-			return (null != role && role === 'admin');
-		});
+	return Promise.reject({
+		status: 403,
+		type: 'missing-roles',
+		message: 'The user does not have the required roles for the team'
+	});
+};
 
-		if (adminFound) {
-			return Promise.resolve();
-		}
-		return Promise.reject({ status: 400, type: 'bad-request', message: 'Team must have at least one admin' });
+/**
+ * Checks if user role meets or exceeds the requestedRole according to
+ * a pre-defined role hierarchy
+ *
+ * @returns {boolean}
+ */
+const meetsOrExceedsRole = (userRole, requestedRole) => {
+	if (null != userRole && _.has(teamRolesMap, userRole) && null != requestedRole && _.has(teamRolesMap, requestedRole)) {
+		return (teamRolesMap[userRole].priority >= teamRolesMap[requestedRole].priority);
 	}
+	return false;
+};
 
-	/**
-	 * Validates that the roles are one of the accepted values
-	 */
-	function validateTeamRole(role) {
-		if (-1 !== teamRoles.indexOf(role)) {
-			return Promise.resolve();
-		}
-
-		return Promise.reject({ status: 400, type: 'bad-argument', message: 'Team role does not exist' });
+/**
+ * Gets the team role for the specified user
+ * and also applies the business logic of if they are implicitly a member
+ * of the team or if they are an inactive explicit member of a team
+ *
+ * @returns Returns a role, or null if the user is not a member of the team
+ */
+const getActiveTeamRole = (user, team) => {
+	if (user && user.constructor.name === 'model') {
+		user = user.toObject();
 	}
 
-	/**
-	 * Creates a new team with the requested metadata
-	 *
-	 * @param teamInfo
-	 * @param creator The user requesting the create
-	 * @returns {Promise} Returns a promise that resolves if team is successfully created, and rejects otherwise
-	 */
-	async function createTeam(teamInfo, creator, firstAdmin, headers) {
-		// Create the new team model
-		const newTeam = new Team();
+	// No matter what, we need to get these
+	const teamRole = getTeamRole(user, team);
 
-		copyMutableFields(newTeam, teamInfo);
+	// User has an explicit team role
+	if (teamRole) {
+		return teamRole;
+	}
 
-		// Write the auto-generated metadata
-		newTeam.creator = creator;
-		newTeam.created = Date.now();
-		newTeam.updated = Date.now();
-		newTeam.creatorName = creator.name;
+	const implicitMembersEnabled = _.get(config, 'teams.implicitMembers.strategy', null) !== null;
 
-		let user = await User.findById(firstAdmin).exec();
-		user = User.filteredCopy(user);
+	// implicit team members is not enabled, or the team does not have implicit members enabled
+	if (!implicitMembersEnabled || !team.implicitMembers) {
+		// Return the team role
+		return teamRole;
+	}
 
-		// Audit the creation action
-		await auditService.audit('team created', 'team', 'create', TeamMember.auditCopy(creator), Team.auditCopy(newTeam), headers);
```

Comment:
@jrassa is `auditService` an unused variable now?

Reply:
fixed

Reply:
fixed